src/staticdata.c

```diff
@@ -718,9 +888,32 @@ static uintptr_t _backref_id(jl_serializer_state *s, jl_value_t *v) JL_NOTSAFEPO
         uint8_t u8 = *(uint8_t*)v;
         return ((uintptr_t)TagRef << RELOC_TAG_OFFSET) + u8 + 2 + NBOX_C + NBOX_C;
     }
+    if (s->incremental && jl_object_in_image(v)) {
+        assert(link_ids);
+        uintptr_t item = add_external_linkage(s, v, link_ids);
+        if (!item) {
+            // If we got here, we failed to find the expected external linkage. Print debugging info.
+            for (size_t i = 0; i < jl_linkage_blobs.len; i+=2) {
+#if defined(__i386__)
+                jl_printf(JL_STDOUT, "i = %d: %p to %p\n", i>>1, jl_linkage_blobs.items[i], jl_linkage_blobs.items[i+1]);
+#else
+                jl_printf(JL_STDOUT, "i = %ld: %p to %p\n", i>>1, jl_linkage_blobs.items[i], jl_linkage_blobs.items[i+1]);
+#endif
```

Comment:
I don't think `__i386__` is a good check, the problem is the word size, not the specific architecture, you'd have the same problem also on armv6l/armv7l.  I think you want to use the `z` format specifier instead
```suggestion
                jl_printf(JL_STDOUT, "i = %zd: %p to %p\n", i>>1, jl_linkage_blobs.items[i], jl_linkage_blobs.items[i+1]);
```